control-plane/api/mesh/v2beta1/gateway_class_config_types.go

```diff
@@ -38,115 +25,79 @@ type GatewayClassConfig struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   pbmesh.GatewayClassConfig `json:"spec,omitempty"`
+	Spec   GatewayClassConfigSpec `json:"spec,omitempty"`
 	Status `json:"status,omitempty"`
 }
 
-// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen=true
 
-// GatewayClassConfigList contains a list of GatewayClassConfig.
-type GatewayClassConfigList struct {
-	metav1.TypeMeta `json:",inline"`
-	metav1.ListMeta `json:"metadata,omitempty"`
-	Items           []*GatewayClassConfig `json:"items"`
-}
+// GatewayClassConfigSpec specifies the desired state of the Config CRD.
+type GatewayClassConfigSpec struct {
 
-func (in *GatewayClassConfig) ResourceID(namespace, partition string) *pbresource.ID {
-	return &pbresource.ID{
-		Name: in.Name,
-		Type: pbmesh.GatewayClassConfigType,
-		Tenancy: &pbresource.Tenancy{
-			Partition: partition,
-			Namespace: namespace,
-
-			// Because we are explicitly defining NS/partition, this will not default and must be explicit.
-			// At a future point, this will move out of the Tenancy block.
-			PeerName: constants.DefaultConsulPeer,
-		},
-	}
-}
+	// +kubebuilder:validation:Enum=ClusterIP;NodePort;LoadBalancer
+	ServiceType *corev1.ServiceType `json:"serviceType,omitempty"`
 
-func (in *GatewayClassConfig) Resource(namespace, partition string) *pbresource.Resource {
```

Comment:
do we still need all these functions like `ResourceID`, `Resource`, etc to maintain compatibility with other v2 resources?

Reply:
I intentionally removed these so that no kubebuilder-based `GatewayClassConfig` can ever accidentally be used as a Consul resource and will produce a compile-time error instead of a runtime error